### PR TITLE
fix jsx symbol collisions when importing own variables with same names

### LINKED ISSUE #14021: Bun 1.1.28 -> ReferenceError: Can't find variable: Fragment
### What version of Bun is running?

1.1.28+cf4e9cb69

### What platform is your computer?

Darwin 23.6.0 arm64 arm

### What steps can reproduce the bug?

In the new version of Bun, it doesn't detect the Fragment variable even if I have it in an import directly from JSX-runtime.

### What is the expected behavior?

That it uses correctly the direct fragment of the import. It crashes in the test and also in build who uses this component that has this import, for something the build is not done well.

```tsx
import { Fragment } from '@/jsx-runtime';

// This now shows an error:
return <Fragment>foo</Fragment>
```

### What do you see instead?

I see this error:

```sh
 97 |         >
 98 |           {content}
 99 |           {style.length > 0 && <style>{toInline(style)}</style>}
100 |         </template>
101 |       )}
102 |       <Fragment slot="">{props.children}</Fragment>
                                                ^
ReferenceError: Can't find variable: Fragment
```

### Additional information

https://github.com/user-attachments/assets/e4ad8643-bfbf-4f94-b414-166ffa3bbd4f

In this video is better to see the error.

### PR DESCRIPTION
**Pull request by snoglobe**
### What does this PR do?

fixes #14021 - importing and using something that shares a name with a JSX symbol caused an import collision and a strange reference error. underlying cause was the symbol collision was not properly handled (plus some legacy code that needed to be removed). 

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### CODE REVIEW

**Comment by robobun** [comment]
@snoglobe, your commit 8a8f7c1f86ea7ca2ccc4e7a81970b06b81514980 is building: [`#4365`](https://buildkite.com/bun/bun/builds/4365)

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
test failures

**Comment by snoglobe** [comment]
fixed

File: src/js_parser.zig [hunk 1]
```diff
@@ -16374,7 +16269,8 @@ fn NewParser_(
                                         break :tagger p.jsxStringsToMemberExpression(expr.loc, p.options.jsx.fragment) catch unreachable;
                                     }
 
-                                    break :tagger p.jsxImport(.Fragment, expr.loc);
+                                    const I_FUCKING_HATE_LLDB = p.jsxImport(.Fragment, expr.loc);
```
Comment (paperclover) [inline src/js_parser.zig]: inline constant into single usage. tbh the label can alos just be named tag to match

**Review by paperclover — CHANGES_REQUESTED** [review]

**Comment by paperdave** [comment]
reasonable otherwise.

File: src/js_parser.zig (hunk 1 above)
Comment (Jarred-Sumner) [reply src/js_parser.zig]: Please don't name variables like this in Bun's codebase.

**Comment by snoglobe** [comment]
forgot to remove it 😓

**Comment by Jarred-Sumner** [comment]
merge conflicts again

**Comment by snoglobe** [comment]
fixed

**Review by paperclover — APPROVED** [review]

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
bundler npm test is failing.
image

**Comment by fel1x-developer** [comment]
Any updates? This bug is blocking some projects to upgrading bun from <1.1.28 , such as honojs.

**Comment by Jarred-Sumner** [comment]
This also needs to update the runtime transpiler cache version before it can be merged.

**Comment by snoglobe** [comment]
fixed

**Pull request merged**